stores/cachestore/cache_store.go

```diff
@@ -0,0 +1,49 @@
+package cachestore
+
+import (
+	"os"
+	"time"
+
+	"github.com/mailchain/mailchain/encoding"
+	"github.com/mailchain/mailchain/internal/hash"
+
+	"github.com/spf13/afero"
+)
+
+type CacheStore struct {
+	cache afero.Fs
+}
+
+func NewCacheStore(cacheTimeout time.Duration, basePath string) *CacheStore {
```

Comment:
Just found out that if the directory does not exist then it will fail when trying to store the message as it can't write the file. Try using something like `fs.MkdirAll` either here or on write. I think on `New` is better otherwise you'd need it on read/write

Reply:
That's totally make sense! I will create the `basePath`